backend/controller/state/state.go

```diff
@@ -0,0 +1,139 @@
+package state
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strconv"
+
+	"github.com/joho/godotenv"
+	log "github.com/sirupsen/logrus"
+)
+
+// State is the global state for the controller.
+type State struct {
+	Log      *log.Logger // Log is a structured event logger
+	IsDocker bool        // IsDocker indicates if running inside Docker
+	Config   *Config     // Config contains global configuration
+	MQTT     *MQTT
+	MQTTCh   chan MQTTMessage
+	AMQP     [2]*AMQP
```

Comment:
Spawning two AMQP channels.

AMQP channels are half duplex. Since our AMQP consumers and producers will be running in separate goroutines, it's easier to have to separate channels over a connection.

Although we can just introduce a mutex, we'll then need to have application logic to route messages to the correct goroutine.